adapter/thing/thermostat_test.go

```diff
@@ -0,0 +1,563 @@
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:thermostat/ad:2", "thermostat"),
+						},
+					},
+					{
+						Name:    "Missing unit when setting setpoint",
+						Command: suite.StringMapMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:thermostat/ad:2", "cmd.setpoint.set", "thermostat", map[string]string{"type": "test_mode_a", "temp": "20.0"}),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:thermostat/ad:2", "thermostat"),
+						},
+					},
+					{
+						Name:    "Missing temperature when setting setpoint",
+						Command: suite.StringMapMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:thermostat/ad:2", "cmd.setpoint.set", "thermostat", map[string]string{"type": "test_mode_a", "unit": "C"}),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:thermostat/ad:2", "thermostat"),
+						},
+					},
+					{
+						Name:    "Missing mode when setting setpoint",
+						Command: suite.StringMapMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:thermostat/ad:2", "cmd.setpoint.set", "thermostat", map[string]string{"temp": "20.0", "unit": "C"}),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:thermostat/ad:2", "thermostat"),
+						},
+					},
+					{
+						Name:    "Invalid temperature when setting setpoint",
+						Command: suite.StringMapMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:thermostat/ad:2", "cmd.setpoint.set", "thermostat", map[string]string{"type": "test_mode_a", "temp": "20C", "unit": "C"}),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:thermostat/ad:2", "thermostat"),
+						},
+					},
+					{
+						Name:    "Setting setpoint with wrong type of value",
+						Command: suite.StringMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:thermostat/ad:2", "cmd.setpoint.set", "thermostat", "22"),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:thermostat/ad:2", "thermostat"),
+						},
+					},
+					{
+						Name:    "Setting setpoint of non-existent thing",
+						Command: suite.StringMapMessage("pt:j1/mt:cmd/rt:dev/rn:test_adapter/ad:1/sv:thermostat/ad:3", "cmd.setpoint.set", "thermostat", map[string]string{"type": "test_mode_a", "temp": "21.0", "unit": "C"}),
+						Expectations: []*suite.Expectation{
+							suite.ExpectError("pt:j1/mt:evt/rt:dev/rn:test_adapter/ad:1/sv:thermostat/ad:3", "thermostat"),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	s.Run(t)
+}
+
+func TestTaskThermostat(t *testing.T) { //nolint:paralleltest
+	s := &suite.Suite{
+		Cases: []*suite.Case{
+			{
+				Name: "Thermostat tasks",
```

Comment:
the more I look at this test, the more I'd like to split it into smaller tests. I mean, we can keep this one to make sure there are no weir interferences between particular tasks, but first I'd test them separately to communicate their purpose clearly. What do you think?

Reply:
Like I explained to you before, because on startup adapter will send all possible states and measurements, I would have to mock calls that are not then checked multiple times, which is in my opinion worse.

Reply:
You can always hide that calls behind a helper method, if they're not relevant from the test perspective.

Reply:
Too much automagic.

Reply:
Ok, let's leave it as it is.